### DISCUSSION TITLE
Using another Database instead of MySQL

### DISCUSSION THREAD

**Discussion by Balachandar-Ganesan** [Leantime/leantime#1043] [Q&A]
Hi Want to port LeanTime from PHP/MySQL into PHP/Another back end either NoSQL,Postgres,SingleStore. Where I do I get the current script and the model. Seems like complete installation needed for creating the needed tables. Any work around to get MySQL scripts for LeanTime.IO

Regards
Bala Ganesan.

**Comment by gloriafolaron** [comment]
@marcelfolaron I'm not aware of anything we have to help support this at this time. FYI here in case you haven't seen it.

**Comment by marcelfolaron** [comment]
Hey,
We currently do not have support for other dbs (nosql, postgres etc)
Postgres might be the closest db that could work with just a few updates.
I don't see how we would get nosql to work without significant rework.
Happy to discuss if there are adapters or similar that could make that work.

The entire script can be found in
https://github.com/Leantime/leantime/blob/1465c70c7b5881e89712c67502fbd85058f86cc0/src/domain/install/repositories/class.install.php#L306
There are a few variables that are set in that call but this should give
you a starting point.

**Reply by facorazza** [reply]
postgres support would be great

**Reply by FreeSynergy** [reply]
Is somebody working on it?

**Reply by marcelfolaron** [reply]
We started by replacing the main db class to use Laravel's database connection handler which is complete, however the repositories now need to be updated to use query builders instead of the custom sql queries to make them work across databases (Ideally we can move to doctrine). Hoping I can throw some AI at it to do the bulk of the rewrites but could definitely need some help here.

**Reply by FreeSynergy** [reply]
> We started by replacing the main db class to use Laravel's database connection handler which is complete, however the repositories now need to be updated to use query builders instead of the custom sql queries to make them work across databases (Ideally we can move to doctrine). Hoping I can throw some AI at it to do the bulk of the rewrites but could definitely need some help here.

How far is the refactoring? Is it possible, to test it?

**Reply by marcelfolaron** [reply]
The db service was replaced with laravel however the queries were not.

They would need to be replaced using query builders or only use common sql functions. Given all the other priorities we just didn't have the time to make all those changes. 

I'd be happy to coordinate if we can find 2 or 3 additional contributors who would like to help with this.

**Comment by Balachandar-Ganesan** [comment]
Thanks a lot for the reply. SingleStore is not a NoSQL product. It is a SQL
based product only. Distributed DB. Liked what you have done. But wanted to
port the same to Singlestore as a part of the hackathon it is conducting,

Regards
Bala Ganesan.
